infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -778,6 +778,137 @@ class PublicationService @Autowired constructor(
         return asCsvFile(orderedPublishedItems, timeZone ?: ZoneId.of("UTC"))
     }
 
+    private fun diffTrackNumber(newTrackNumber: TrackLayoutTrackNumber, oldTrackNumber: TrackLayoutTrackNumber?) =
+        listOf(
+            if (oldTrackNumber == null || newTrackNumber.number != oldTrackNumber.number) {
```

Comment:
Siistimisajatuksena:
```
fun <T> changeOf(name: String, oldValue: T, newValue: T, remark: PublicationChangeRemark? = null): PublicationChange? =
  if (oldValue == newValue) null
  else PublicationChange(name, oldValue, newValue, remark)
```
Tuolla rakenteella nämä listausten haut menisi listOfNotNull:lla yksirivisiksi:
```
listOfNotNull(
  changeOf("track-number", old..., new...),
  changeOf(...
)
```

Tuosta voisi tietty jalostaa eteenpäin vielä toisen funkkarin jolle annetaan oliot ja getteri kenttäarvojen sijaan ja joka puolestaan kutsuu tuota... mutta se ei ehkä maksa vaivaa jos olioiden fetch muuttuu muutenkin.

Reply:
Lopun filtteröinnin voi toki korvata listOfNotNull:lla ilmankin että tekee tuota muuta remppaa